packages/stark-core/src/common/bootstrap/abstract-stark-main.ts

```diff
@@ -0,0 +1,163 @@
+		document.removeEventListener("DOMContentLoaded", this.mainWrapper, false);
+
+		this.initialize();
+	};
+
+	/**
+	 * Call the main() method and log an error in case of failure
+	 */
+	private invokeMain(): void {
+		// FIXME handle errors nicely (UX!)
+		this.main().catch((err: any) => console.error(err));
+
+		// TODO reintroduce later
+		// bootstrap the app as long as the app root element is there (it won't be there in case the browser is outdated)
+		// if (document.getElementsByClassName("stark-app")[0]) {
+		// 	// add anything relevant here
+		// } else {
+		// 	console.error("Could not bootstrap the App: unsupported browser.");
+		// }
+	}
+
+	/**
+	 * Initialize by configuring HMR if it is enabled or normally instead.
+	 */
+	protected initialize = (): void => {
+		if (ENV === "development" && this.environment.hmr) {
+			if (module["hot"]) {
+				this.bootstrapHmr(module, this.main);
+			} else {
+				console.error("HMR is not enabled for webpack-dev-server! This is most probably due to a bug in Stark.");
+			}
+		} else {
+			this.invokeMain();
+		}
+	};
+
+	/**
+	 * Call the main function once the DOM has loaded
+	 */
+	protected bootstrapDomReady = (): void => {
+		switch (document.readyState) {
+			case "loading":
+				document.addEventListener("DOMContentLoaded", this.mainWrapper, false);
+				break;
+			case "interactive":
+			case "complete":
+			default:
+				this.initialize();
+		}
+	};
+
+	/**
+	 * Bootstrap after document is ready
+	 */
+	public bootstrap(): void {
+		console.log(`
+NNNNNNNN        NNNNNNNNBBBBBBBBBBBBBBBBB   BBBBBBBBBBBBBBBBB   
+N:::::::N       N::::::NB::::::::::::::::B  B::::::::::::::::B
+N::::::::N      N::::::NB::::::BBBBBB:::::B B::::::BBBBBB:::::B
+N:::::::::N     N::::::NBB:::::B     B:::::BBB:::::B     B:::::B
+N::::::::::N    N::::::N  B::::B     B:::::B  B::::B     B:::::B
+N:::::::::::N   N::::::N  B::::B     B:::::B  B::::B     B:::::B
+N:::::::N::::N  N::::::N  B::::BBBBBB:::::B   B::::BBBBBB:::::B
+N::::::N N::::N N::::::N  B:::::::::::::BB    B:::::::::::::BB
+N::::::N  N::::N:::::::N  B::::BBBBBB:::::B   B::::BBBBBB:::::B
+N::::::N   N:::::::::::N  B::::B     B:::::B  B::::B     B:::::B
+N::::::N    N::::::::::N  B::::B     B:::::B  B::::B     B:::::B
+N::::::N     N:::::::::N  B::::B     B:::::B  B::::B     B:::::B
+N::::::N      N::::::::NBB:::::BBBBBB::::::BBB:::::BBBBBB::::::B
+N::::::N       N:::::::NB:::::::::::::::::B B:::::::::::::::::B
+N::::::N        N::::::NB::::::::::::::::B  B::::::::::::::::B
+NNNNNNNN         NNNNNNNBBBBBBBBBBBBBBBBB   BBBBBBBBBBBBBBBBB
+
+We need great software developers like you! https://jobs.nbb.be
+		`);
+
+		this.bootstrapDomReady();
+	}
+
+	/**
+	 * Configure HMR
+	 * Code based on: https://github.com/angular/angular-cli/wiki/stories-configure-hmr
+	 * Reference: https://github.com/gdi2290/angular-hmr
+	 * @ignore
+	 */
+	protected bootstrapHmr: Function = (module: any, bootstrap: () => Promise<NgModuleRef<any>>) => {
+		if(ENV === "development") {
```

Comment:
This is already checked at line 51. Maybe could we remove this check ? 😊

Reply:
Depends on how intelligent Webpack is for dead code removal. I didn't check the generated code though, I've added the check in the function just be be sure that the code would be dropped in production builds

Reply:
Indeed, as @dsebastien said, this is needed so that webpack can remove the code inside this function.

The other check (in line 51) makes sure that this function will not be called, but yet the function will still be there and so code that calls the "angularclass/hmr" module.